src/services/goToDefinition.ts

```diff
@@ -40,15 +40,15 @@ namespace ts.GoToDefinition {
             });
         }
 
-        let { symbol, failedAliasResolution } = getSymbol(node, typeChecker);
+        let { symbol, failedAliasResolution } = getSymbol(node, typeChecker, skipAlias);
         let fallbackNode = node;
 
         if (searchOtherFilesOnly && failedAliasResolution) {
             // We couldn't resolve the specific import, try on the module specifier.
             const importDeclaration = forEach([node, ...symbol?.declarations || emptyArray], n => findAncestor(n, isAnyImportOrBareOrAccessedRequire));
             const moduleSpecifier = importDeclaration && tryGetModuleSpecifierFromDeclaration(importDeclaration);
             if (moduleSpecifier) {
-                ({ symbol, failedAliasResolution } = getSymbol(moduleSpecifier, typeChecker));
+                ({ symbol, failedAliasResolution } = getSymbol(moduleSpecifier, typeChecker, skipAlias));

```

Comment:
@andrewbranch I'd like you to take a look at this case and the interaction between `searchOtherFilesOnly` and the new `skipAlias`

Reply:
`skipAlias` should always be `true` when `serachOtherFilesOnly` is `true`.